aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageDownloadFileOperation.java

```diff
@@ -76,11 +81,20 @@ public void start() {
             return;
         }
 
-        String serviceKey = S3Keys.createServiceKey(
-                getRequest().getAccessLevel(),
-                getRequest().getTargetIdentityId(),
-                getRequest().getKey()
-        );
+        String serviceKey;
+
+        try {
+            serviceKey = S3Keys.createServiceKey(
+                    getRequest().getAccessLevel(),
+                    getRequest().getTargetIdentityId() != null
+                            ? getRequest().getTargetIdentityId()
+                            : awsAuthProvider.getIdentityId(),
+                    getRequest().getKey()
+            );
+        } catch (StorageException exception) {
```

Comment:
Can you scope this to only the new auth lines that throw it?

```
final String maybeVar;
try {
   maybeVar = getIt();
} catch (StorageException varNotAvailable) {
   blowUp();
}
// maybeVar is now a legit var.
```

Reply:
Sure